### PR TITLE
Add Symfony sample for PHP 7.2 on App Engine Standard

### PR DESCRIPTION
**Pull request by bshaffer**
Would love a review from @grayside if you're bored!!

### CODE REVIEW

File: appengine/php72/symfony-framework/README.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        APP_SECRET: YOUR_APP_SECRET
+
+    # URL handlers
+    # ...
+    ```
+
+1. Set the cache and log directories to `/tmp` in production. This is because
+   App Engine's file system is **read-only**. Modify the functions `getCacheDir`
+   and `getLogDir` to the following in `src/Kernel.php`:
+
```
Comment (grayside) [inline appengine/php72/symfony-framework/README.md]: Might want to link to https://symfony.com/doc/current/configuration/override_dir_structure.html or https://symfony.com/doc/4.1/configuration/override_dir_structure.html

File: appengine/php72/symfony-framework/README.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+   with the name of your instance:
+
+       gcloud sql databases create symfonydb --instance=INSTANCE_NAME
+
+1. Enable the [CloudSQL APIs][cloudsql-apis] in your project.
+
+1. Follow the instructions to
+   [install and run the Cloud SQL proxy client on your local machine][cloudsql-install].
+   The Cloud SQL proxy is used to connect to your Cloud SQL instance when
+   running locally.
```
Comment (grayside) [inline appengine/php72/symfony-framework/README.md]: We are recommended developers connect to the prod db from local? Maybe this should demonstrate creating a second dev DB?

Comment (grayside) [reply appengine/php72/symfony-framework/README.md]: Oh, reading on this is so you can do some database setup via code tools. Might want to mention that up above.

File: appengine/php72/symfony-framework/README.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+1.  Modify your `app.yaml` file with the following contents. Be sure to replace
+    `DB_PASSWORD` and `INSTANCE_CONNECTION_NAME` with the values you created for
+    your Cloud SQL instance:
+
+    ```yaml
+    runtime: php72
+
+    env_variables:
+      APP_ENV: prod
+      APP_SECRET: YOUR_APP_SECRET
```
Comment (grayside) [inline appengine/php72/symfony-framework/README.md]: I don't see what the secret is for in the README.

File: appengine/php72/symfony-framework/app.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    static_dir: public/build
+  - url: /bundles
+    static_dir: public/bundles
+
+  # Declare any media files in the public directory as static assets as well.
+  - url: /(.*\.(ico|txt|gif|png|jpg))$
+    static_files: public/\1
+    upload: public/.*\.(ico|txt|gif|png|jpg)$
+
+  # Declare a catch-all to send all other routes to be handled by the application.
```
Comment (grayside) [inline appengine/php72/symfony-framework/app.yaml]: Why do you need to do this?

File: appengine/php72/symfony-framework/composer.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "require-dev": {
+        "google/cloud-tools": "^0.8.2",
+        "monolog\/monolog": "^1.19",
+        "nikic/php-parser": "^4.0"
+    }
+}
```
Comment (grayside) [inline appengine/php72/symfony-framework/composer.json]: Where did symfony go? If you are using the symfony demo template, it's this: https://github.com/symfony/demo/blob/master/composer.json

If the point is you create the scaffolding/take your project, and make sure to add the above dependencies some more callout in the README seems necessary.

Likewise, without more symfony pieces how are there tests testing?

File: appengine/php72/symfony-framework/test/DeployTest.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright 2016 Google Inc.
```
Comment (grayside) [inline appengine/php72/symfony-framework/test/DeployTest.php]: Time travel to the world of 2016

File: appengine/php72/symfony-framework/README.md (hunk 3 above)
Comment (bshaffer) [reply appengine/php72/symfony-framework/README.md]: It's a required Symfony parameter. I'll add [a link to the Symfony docs](http://symfony.com/doc/current/reference/configuration/framework.html#secret). Is that sufficient?

File: appengine/php72/symfony-framework/app.yaml (hunk 4 above)
Comment (bshaffer) [reply appengine/php72/symfony-framework/app.yaml]: Haha ho boy wouldnt you like to know. I'll add you in on the great internal debate...

File: appengine/php72/symfony-framework/README.md (hunk 1 above)
Comment (bshaffer) [reply appengine/php72/symfony-framework/README.md]: Good idea!

File: appengine/php72/symfony-framework/composer.json (hunk 5 above)
Comment (bshaffer) [reply appengine/php72/symfony-framework/composer.json]: This `composer.json`, as you correctly identified, is just for testing the steps in the README. The actual instance of symfony is deployed in our tests [here](https://github.com/GoogleCloudPlatform/php-docs-samples/blob/php72-symfony/appengine/php72/symfony-framework/test/lib/SymfonyDeploymentTrait.php#L53).

I'm not attached to this method of doing it, but it's the best I've come up with 😄

File: appengine/php72/symfony-framework/test/lib/SymfonyDeploymentTrait.php [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            {
+                $this->newStmts = $newStmts;
+            }
+
+            public function enterNode(Node $node)
+            {
+                if ($node instanceof ClassMethod
+                    && in_array($node->name, ['getCacheDir', 'getLogDir'])) {
+                    $node->stmts = array_merge($this->newStmts, $node->stmts);
+                }
```
Comment (grayside) [inline appengine/php72/symfony-framework/test/lib/SymfonyDeploymentTrait.php]: Wow! 😲 

Haven't seen AST for code merge before, nice trick!

Does this continue to work work if the symfony-demo template adds an implementation of these functions?

Comment (bshaffer) [reply appengine/php72/symfony-framework/test/lib/SymfonyDeploymentTrait.php]: The `getCacheDir` and `getLogDir` functions are already implemented, the code here just sticks the `if ($this->environment === 'prod')...` conditional at the top of the functions. I could have just done something like 

```php
$newConditional = "if ($this->environment === 'prod'...)";
$newCode = str_replace("function getCacheDir()\n\t{", "function getCacheDir()\n\t{" . $newConditional, $oldCode);
```

But this way should be cleaner / less brittle.

**Pull request merged**